### PR TITLE
Enable auto sign in after user makes a selection on a credential in the hint and retrieve flows.

### PR DESCRIPTION
**Pull request by tikurahul**

### CODE REVIEW

File: ts/spi/provider_frame_test.ts [hunk 1]
```diff
@@ -310,9 +310,12 @@ describe('ProviderFrame', () => {
                      return Promise.resolve();
                    });
 
-           clientChannel.listen(msg.RpcMessageType.credential, (data) => {
+           clientChannel.listen(msg.RpcMessageType.credential, async (data) => {
              expectMessageContents(
                  data, msg.credentialResultMessage(requestId, alicePwdCred));
+             let enabled =
```
Comment (TMSCH) [inline ts/spi/provider_frame_test.ts]: const

File: ts/spi/provider_frame_test.ts [hunk 2]
```diff
@@ -398,8 +401,11 @@ describe('ProviderFrame', () => {
                      return Promise.resolve(alicePwdCred);
                    });
 
-           clientChannel.listen(msg.RpcMessageType.credential, (data) => {
+           clientChannel.listen(msg.RpcMessageType.credential, async (data) => {
              expect(expectFinalResult).toBeTruthy();
+             let enabled =
```
Comment (TMSCH) [inline ts/spi/provider_frame_test.ts]: const

File: ts/spi/provider_frame_test.ts [hunk 3]
```diff
@@ -691,6 +695,12 @@ describe('ProviderFrame', () => {
             [elisaOtherDomainCred],
             elisaOtherDomainCred,
             msg.credentialResultMessage(requestId, redactedElisaCred))
+            .then(() => {
+              return localStateProvider.isAutoSignInEnabled(TEST_AUTH_DOMAIN);
+            })
+            .then((enabled) => {
+              expect(enabled).toBe(true);
+            })
             .then(done);
```
Comment (TMSCH) [inline ts/spi/provider_frame_test.ts]: nit: the done doesn't have to be in its own callback, you could move it in the function's body before.

Comment (tikurahul) [inline ts/spi/provider_frame_test.ts]: Done

**Pull request merged**